Library/Homebrew/hardware.rb

```diff
@@ -41,6 +41,20 @@ def optimization_flags
       end
       alias generic_optimization_flags optimization_flags
 
+      # Rust already defaults to the oldest supported cpu for each target-triple
+      # so it's safe to ignore generic archs such as :armv6, etc., here.
+      # Rust defaults to apple-m1 since Rust 1.71 for aarch64-apple-darwin.
+      def rust_optimisation_flags
+        @rust_optimisation_flags ||= {
+          native:      "--codegen target-cpu=native",
+          ivybridge:   "--codegen target-cpu=ivybridge",
+          sandybridge: "--codegen target-cpu=sandybridge",
+          nehalem:     "--codegen target-cpu=nehalem",
+          core2:       "--codegen target-cpu=core2",
+          core:        "--codegen target-cpu=prescott",
```

Comment:
```suggestion
        @rust_optimisation_flags ||= {
          native:      "--codegen target-cpu=native",
          ivybridge:   "--codegen target-cpu=ivybridge",
          sandybridge: "--codegen target-cpu=sandybridge",
          nehalem:     "--codegen target-cpu=nehalem",
          core2:       "--codegen target-cpu=core2",
          core:        "--codegen target-cpu=prescott",
```
Feels like an array of symbols would be more appropriate here and turning that into `--codegen target-cpu` should be part of `rustflags_target_cpu`?